api_spec.yaml

```diff
@@ -3246,9 +3246,497 @@ paths:
+                      $ref: '#/components/schemas/tenantId'
+        '400':
+          $ref: '#/components/responses/400'
+
+        '401':
+          $ref: '#/components/responses/401'
+
+        '404':
+          $ref: '#/components/responses/404'
+
+        '500':
+          $ref: '#/components/responses/500'
+
+  /recipe/multitenancy/config/thirdparty:
+    put:
+      tags:
+        - Multitenancy Recipe
+      description: |
+        Creates or updates ThirdParty Provider Config for a tenant.
+
+        If `tenantId` is not provided, `defaultTenantId` will be used.
+
+        If `tenantId` is specified and is not equal to `defaultTenantId`, SuperTokens subscription license key is required.
+
+        If `skipValidation` is set to true, the config will be saved without any validation.
+
+        **Validations done by the core:**
+
+        `thirdPartyId` and `name` are required
+
+        Common to all providers:
+          - `clients` list:
+            - can be undefined or empty
+            - if contains 1 element, clientType can be empty or undefined
+            - if contains more than 1 element, clientType must be defined and unique
+          - for each element in `clients`:
+            - `clientId` must not be empty
+
+        Apple:
+          - `clients`
+            - if it contains elements, each of them are validated as follows:
+              - `clientSecret` must be empty or undefined
+              - `additionalConfig` should contain the following keys - `keyId`, `teamId`, `privateKey`
+                - `keyId` must be a non empty string
+                - `teamId` must be a non empty string
+                - `privateKey` must be a non empty string
+
+        Active Directory:
+          - `clients`
+            - if it contains elements, each of them are validated as follows:
+              - `additionalConfig` should contain the key `directoryId`
+                - `directoryId` must be a non empty string
+
+        Okta:
+          - `clients`
+            - if it contains elements, each of them are validated as follows:
+              - `additionalConfig` should contain the key `oktaDomain`
+                - `oktaDomain` must be a non empty string
+
+        Google Workspaces:
+          - `clients`
+            - if it contains elements, each of them are validated as follows:
+              - `additionalConfig` may contain the key `hd`
+                - `hd` is optional
+                - if specified, it must be either `"*"`, or a valid domain
+
+        Discord:
+          - `clients`
+            - if it contains elements, each of them are validated as follows:
+              - `additionalConfig` may contain the key `requireEmail`
+                - `requireEmail` is optional and defaults to `false`
+                - if specified, it must be a `boolean`
+
+        Boxy SAML
+          - `clients`
+            - `additionalConfig` in the each element must contain `boxyURL` and must be non-empty string
+
```

Comment:
modify the description to add what happens if a config already exists for a thirdpartyid and tenantId